data/net/minecraft/world/level/levelgen/feature/Feature.mapping

```diff
@@ -20,11 +26,24 @@ CLASS net/minecraft/world/level/levelgen/feature/Feature
 		ARG 0 blockTag
 	METHOD isStone (Lnet/minecraft/world/level/block/state/BlockState;)Z
 		ARG 0 state
+	METHOD lambda$isReplaceable$2 (Lnet/minecraft/world/level/block/state/BlockState;)Z
+		ARG 0 state
+	METHOD lambda$isReplaceable$3 (Lnet/minecraft/tags/Tag;Lnet/minecraft/world/level/block/state/BlockState;)Z
+		ARG 1 state
+	METHOD lambda$new$0 (Lnet/minecraft/world/level/levelgen/feature/configurations/FeatureConfiguration;)Lnet/minecraft/world/level/levelgen/feature/ConfiguredFeature;
+		ARG 1 config
+	METHOD lambda$new$1 (Lnet/minecraft/world/level/levelgen/feature/ConfiguredFeature;)Lnet/minecraft/world/level/levelgen/feature/configurations/FeatureConfiguration;
+		ARG 0 configuredFeature
 	METHOD markAboveForPostProcessing (Lnet/minecraft/world/level/WorldGenLevel;Lnet/minecraft/core/BlockPos;)V
+		COMMENT If the above two blocks are not air, marks them for post processing.
+		COMMENT This is used to prevent floating grass during the generation of features that carve blocks out of the terrain, after other plant-like blocks have generated (such as lake features).
 		ARG 1 level
 		ARG 2 basePos
 	METHOD place (Lnet/minecraft/world/level/levelgen/feature/FeaturePlaceContext;)Z
-		ARG 1 featurePlaceContext
+		COMMENT Places the given feature at the given location.
+		COMMENT During world generation, features are provided with a 3x3 region of chunks that they can safely generate into.
```

Comment:
The 3x3 region of chunks referenced here is centered around the block position within the place context provided, right? It may be a good idea to note that here.

Reply:
> is centered around the block position within the place context provided

No.
Technicality, but it is centered on the `center` chunk position passed in the `WorldGenRegion` (which implements `WorldGenLevel`) found on the `FeaturePlaceContext`. A decorator is absolutely free to move said position outside the generating chunk. (Although in practice this rarely, if ever, happens)